### ISSUE TITLE
Push trigger only works for the multi-branch mode

### ISSUE THREAD

**Issue by joekur** [lumoslabs/txgh#70]
Can you clarify if this was intended, or if I'm missing something? The push trigger endpoint expects a `branch` param, and then it creates a resource tagged with the branch name, even though the project is configured to use the basic workflow.

For context, I am not using multi-branch because my understanding is that it will create a resource any time someone pushes a new branch to github. I'm not sure I like the idea that if someone is starting a WIP feature, and they add a translation file that may not yet be finalized, that it ends up immediately in transifex. Thus our flow for now is that nothing gets translated until it's in master.

**Comment by camertron** [comment]
@joekur the push trigger always requires a branch. In the basic workflow, that branch will be the default, i.e. `master`. It's configured by the `branch: ` config option in your txgh yaml config.

**Comment by joekur** [comment]
Yeah but even if you supply your main branch, you end up with a new resource in transifex. Eg in this scenario, the first resource here is created by a github webhook, the second by a push trigger:

![screen shot 2017-02-23 at 1 16 39 pm](https://cloud.githubusercontent.com/assets/1043396/23275012/70c9a708-f9ca-11e6-88f0-b278db9b3cbc.png)

(note `develop` is being used as the main branch)

**Comment by camertron** [comment]
Ahh ok I see. You'd have to rename the existing resource in order for txgh to recognize it as belonging to the develop branch. Specifically I mean you'd have to change the resource's slug, since the branch information is included in each of them. Is it not feasible for you to change the existing resource slugs?

In any case, it should be fairly straightforward to not include the branch in the resource slug when using the normal workflow. I thought that was already supported (see `Txgh::TxResource` and `Txgh::TxBranchResource`), but I guess it's not working 😞

**Comment by joekur** [comment]
I'm sure it would be feasible, just not really ideal. On the other hand, I wonder if just using the tx client to push and pull would be equivalent if I'm using just the basic mode?

**Comment by joekur** [comment]
I guess it wouldn't update the github statuses though

**Comment by camertron** [comment]
Exactly, you'd get all the benefits except github statuses.
